enclave/core/sgx/tracee.c

```diff
@@ -9,28 +9,36 @@
 #include "../tracee.h"
 #include "report.h"
 
+static volatile int _is_enclave_debug_allowed = -1;
+
 // Read an enclave's identity attribute to see to if it was signed as an debug
 // enclave
 bool is_enclave_debug_allowed()
 {
-    bool ret = false;
-    oe_sgx_td_t* td = oe_sgx_get_td();
+    if (_is_enclave_debug_allowed != -1)
+        goto done;
 
+    oe_sgx_td_t* td = oe_sgx_get_td();
```

Comment:
This could result in compiler warnings since the goto in line 19 bypasses the definition of this variable.